src/views/Pools/components/PoolsTable/Apr.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useMemo } from 'react'
+import { Flex, useModal, CalculateIcon, IconButton, Skeleton, FlexProps } from '@pancakeswap/uikit'
+import { BASE_EXCHANGE_URL } from 'config'
+import ApyCalculatorModal from 'components/ApyCalculatorModal'
+import Balance from 'components/Balance'
+import { useBusdPriceFromToken } from 'state/hooks'
+import { Pool } from 'state/types'
+import { getPoolApr } from 'utils/apr'
+import { getRoi, tokenEarnedPerThousandDollarsCompounding } from 'utils/compoundApyHelpers'
+import { getBalanceNumber } from 'utils/formatBalance'
+import { useTranslation } from 'contexts/Localization'
+
+interface AprProps extends FlexProps {
+  pool: Pool
+  isAutoVault?: boolean
+  performanceFee?: number
+}
+
+const Apr: React.FC<AprProps> = ({ pool, isAutoVault, performanceFee = 0, ...props }) => {
+  const { stakingToken, earningToken, totalStaked, isFinished, tokenPerBlock } = pool
+  const { t } = useTranslation()
+
+  const earningTokenPrice = useBusdPriceFromToken(earningToken.symbol)
+  const earningTokenPriceAsNumber = earningTokenPrice && earningTokenPrice.toNumber()
+  const stakingTokenPrice = useBusdPriceFromToken(stakingToken.symbol)
+  const stakingTokenPriceAsNumber = stakingTokenPrice && stakingTokenPrice.toNumber()
+
+  const apr = !isFinished
+    ? getPoolApr(
+        stakingTokenPriceAsNumber,
+        earningTokenPriceAsNumber,
+        getBalanceNumber(totalStaked, stakingToken.decimals),
+        parseFloat(tokenPerBlock),
+      )
+    : 0
+
+  // special handling for tokens like tBTC or BIFI where the daily token rewards for $1000 dollars will be less than 0.001 of that token
+  const isHighValueToken = Math.round(earningTokenPriceAsNumber / 1000) > 0
+  const roundingDecimals = isHighValueToken ? 4 : 2
```

Comment:
It seems a bit of a shame to have stuff like this repeated from the card `AprRow` component rather than creating something that can be shared? It adds a bit of maintenance to make these changes in multiple components going forward. It's pretty likely that these decimals will need tweaking in future.